realm/realm-library/src/main/java/io/realm/RealmConfiguration.java

```diff
@@ -481,6 +495,17 @@ public Builder rxFactory(RxObservableFactory factory) {
             return this;
         }
 
+        /**
+         * Sets the initial data in {@link io.realm.Realm}. This transaction will be executed only for the first time
+         * when database file is created or while migrating the data when {@link Builder#deleteRealmIfMigrationNeeded()} is set.
+         *
+         * @param transaction transaction to execute.
+         */
+        public Builder initialData(Realm.Transaction transaction) {
```

Comment:
Yes, I agree with @beeender. The idea behind this initial data transaction is to use `version == unversioned` condition (really simple solution). In `DynamicRealm` it is not easy to spot place where we can use this initial dataset transaction.